pkgs/tools/filesystems/dduper/default.nix

```diff
@@ -0,0 +1,51 @@
+{ lib, stdenv, fetchpatch, fetchFromGitHub, btrfs-progs, python3 }:
+
+let
+  btrfsProgsPatched = btrfs-progs.overrideAttrs (oldAttrs: {
+    version = "5.9";
+    patches = [
+        (fetchpatch {
+          name = "0001-Print-csum-for-a-given-file-on-stdout.patch";
+          url = "https://raw.githubusercontent.com/Lakshmipathi/dduper/8fab08e0f1901bf54411d25f1767b48c978074cb/patch/btrfs-progs-v5.9/0001-Print-csum-for-a-given-file-on-stdout.patch";
+          sha256 = "0y8358y3dmr7jsk0qfa1cbpa83ivnjrddwg0nq2gjr7fav860vhf";
+        })
+      ];
+    });
+  py3 = python3.withPackages (ps: with ps;
+    [ prettytable
+      numpy
+    ]);
+in
+stdenv.mkDerivation rec {
+  pname = "dduper";
+  version = "v0.04";
+
+  src = fetchFromGitHub
+    { owner = "lakshmipathi";
+      repo = "${pname}";
+      rev = "${version}";
+      sha256 = "09ncdawxkffldadqhfblqlkdl05q2qmywxyg6p61fv3dr2f2v5wm";
+    };
+
+  buildInputs =
+    [ btrfsProgsPatched
+      py3
+    ];
+
+  patchPhase = ''
+    sed -i -e "s|/usr/sbin/btrfs.static|${btrfsProgsPatched}/bin/btrfs|g" ./dduper;
```

Comment:
please use substituteInPlace instead. It prints a warning when the substitution cannot apply, instead of silently doing nothing